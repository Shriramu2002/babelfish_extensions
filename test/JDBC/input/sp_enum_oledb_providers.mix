-- tsql
-- Executing sp_enum_oledb_providers without tds_fdw extension (should not return anything)
EXEC sp_enum_oledb_providers
GO

-- Creating a simple login which has lesser privilege than sysadmin role
create login linked_server_login_862 with password = 'password_862'
GO

-- tsql    user=linked_server_login_862    password=password_862
-- Executing sp_enum_oledb_providers stored procedure without system admin privilege (should throw error)
EXEC sp_enum_oledb_providers
GO

-- psql
-- Need to terminate active session before cleaning up the login
SELECT pg_terminate_backend(pid) FROM pg_stat_get_activity(NULL) 
WHERE sys.suser_name(usesysid) = 'linked_server_login_862' AND backend_type = 'client backend' AND usesysid IS NOT NULL;
GO

-- tsql
DROP LOGIN linked_server_login_862
GO