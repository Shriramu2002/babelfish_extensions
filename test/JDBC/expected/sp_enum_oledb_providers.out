-- tsql
-- Executing sp_enum_oledb_providers without tds_fdw extension (should not return anything)
EXEC sp_enum_oledb_providers
GO
~~START~~
nvarchar#!#nvarchar#!#nvarchar
TDS_FDW#!#{f34cef3b-778b-23ac-4fff-7d01c089051e}#!#A PostgreSQL foreign data wrapper to connect to TDS databases 2.0.3
~~END~~


-- Creating a simple login which has lesser privilege than sysadmin role
create login linked_server_login with password = 'password'
GO

-- tsql    user=linked_server_login    password=password
-- Executing sp_enum_oledb_providers stored procedure without system admin privilege (should throw error)
EXEC sp_enum_oledb_providers
GO
~~ERROR (Code: 15003)~~

~~ERROR (Message: Only members of the sysadmin role can execute this stored procedure.)~~


-- psql
-- Need to terminate active session before cleaning up the login
SELECT pg_terminate_backend(pid) FROM pg_stat_get_activity(NULL) 
WHERE sys.suser_name(usesysid) = 'linked_server_login' AND backend_type = 'client backend' AND usesysid IS NOT NULL;
GO
~~START~~
bool
t
~~END~~


-- tsql
DROP LOGIN linked_server_login
GO
